AxonDeepSeg/download_model.py

```diff
@@ -1,64 +1,39 @@
+        model_destination = destination / model
 
-    url_sem_destination = "https://github.com/axondeepseg/default-SEM-model/archive/refs/tags/" + sem_release_version + ".zip"
-    url_tem_destination = "https://github.com/axondeepseg/default-TEM-model/archive/refs/tags/" + tem_release_version + ".zip"
-    url_bf_destination = "https://github.com/axondeepseg/default-BF-model/archive/refs/tags/" + bf_release_version + ".zip"
+    url_model_destination = MODELS[model]['weights'][model_type]
 
     files_before = list(Path.cwd().iterdir())
-
-    if (
-        not download_data(url_sem_destination) and not download_data(url_tem_destination) and not download_data(url_bf_destination) 
-    ) == 1:
-        print("Data downloaded and unzipped succesfully.")
+    if not download_data(url_model_destination):
+        print("Model downloaded and unzipped succesfully.")
     else:
-        print(
-            "ERROR: Data was not succesfully downloaded and unzipped- please check your link and filename and try again."
-        )
-
+        print("ERROR: Model was not succesfully downloaded and unzipped- "
+              "please check your link and filename and try again.")
     files_after = list(Path.cwd().iterdir())
 
-    # retrieving unknown model folder names
-    folder_name_SEM_model = Path("default-SEM-model-" + sem_release_version)
-    folder_name_TEM_model = Path("default-TEM-model-" + tem_release_version)
-    folder_name_BF_model = Path("default-BF-model-" + bf_release_version)
-
-    if sem_destination.exists():
-        print('SEM model folder already existed - deleting old one')
-        shutil.rmtree(str(sem_destination))
+    # retrieving unknown model folder name
+    folder_name = list(set(files_after) - set(files_before))[0]
 
-    if tem_destination.exists():
-       print('TEM model folder already existed - deleting old one')
-       shutil.rmtree(str(tem_destination))      
-
-    if bf_destination.exists():
-       print('BF model folder already existed - deleting old one')
-       shutil.rmtree(str(bf_destination))
-
-    shutil.move(folder_name_SEM_model.joinpath("model_seg_rat_axon-myelin_sem"), str(sem_destination))
-    shutil.move(folder_name_TEM_model.joinpath("model_seg_mouse_axon-myelin_tem"), str(tem_destination))
-    shutil.move(folder_name_BF_model.joinpath("model_seg_rat_axon-myelin_bf"), str(bf_destination))
+    if model_destination.exists():
+        print('Model folder already existed - deleting old one')
+        shutil.rmtree(str(model_destination))
+    
+    model_suffix = 'light' if model_type == 'light' else 'ensemble'
+    model_name = f'{MODELS[model]['name']}_{model_suffix}'
+    shutil.move(folder_name.joinpath(model_name), str(model_destination)
 
-    # remove temporary folders
-    shutil.rmtree(folder_name_SEM_model)
-    shutil.rmtree(folder_name_TEM_model)
-    shutil.rmtree(folder_name_BF_model)
+    # remove temporary folder
+    shutil.rmtree(folder_name)
 
 def main(argv=None):
```

Comment:
We need to expose some options for CLI users so that they can download the model they want themselves. The `download_model()` fn now takes 2 important arguments: `model` (e.g. generalist, dedicated-CARS, etc.) and `model_type` (ensemble or light).